docs/design2.md

```diff
@@ -0,0 +1,102 @@
+# Key Transparency 2.0
+
+## Objective
+
+Key Transparency currently sufferes from a design-space tradeoff in two dimentions between
+speed and verification bandwidth.  Each snapshot of the key-value merkle tree creates an additional
+point in time that clients must audit when inspecting their account history. 
+Using Key Transparency in this mode requires the presence of powerful auditors who can verify
+that no information has been lost between all snapshots.
+
+| Update Speed | Trust, then Verifiy | Verify First |
+|--------------|---------------------|--------------|
+| hr           | Tractable Bandwidth | Unusable     |
+| min          |                     | Intractable  |
+| s            |                     | Intractable  |
+
+
+Key Transparency 2.0 is a new set of algorithms that resolves this fundamental tradeoff by sharing 
+the task of auditing between pairs of clients. As a result, all configurations result in efficiently
+auditable data structures, without the presence of 3rd parties. 
+
+We expect that many applications will want to prioritize speed and reliability by using a trust, then verify mode, but because the client operations are very similar, users could have the option to wait a bit longer in order to have high confidence that they are using data from a snapshot that is widely available. 
+
+| Update Speed | Trust, then Verifiy | Verify First |
+|--------------|---------------------|--------------|
+| hr           | Tractable Bandwidth | Unusable     |
+| min          | Tractable Bandwidth | Opt-in       |
+| s            |                     | Tractable    |
+
+
+## Data Structures
+
+### Gossip Network
+The job of the gossip network is to ensure that there is a single, globally consistent lineage of log roots. 
+To keep $n^2$ communication costs low, and to prevent sybil attacks, clients use a small ~20 set of gossip nodes.
+
+Each gossip node fetches the latest signed log root (SLRs) and verifies consistency with all previously seen roots.
+After verifying, the gossip node signs the log root.
+
+Gossip nodes offer the following APIs:
+
+1. Get Signed Root 
+
+### Log of Map Roots
+
+The log server collects signatures from the gossip nodes to produce a Signed Log Root that has a quorum of gossip signatures.
+
+The log server contains a list of map roots representing sequential snapshots of a key-value directory.
+
+Log servers offer the following APIs:
+
+1. Latest Signed Log Root
+1. Get consistency proof between log size a and b. 
+1. Get log item i with inclusion proof to log size a.
+
+### Map Root Snapshots 
+Each Map Root represents a snapshot of a key-value dictionary.
+The map is implemented as a sparse merkle tree of fixed depth. This should be changed to a patricia-prefix-tri for better efficiency.
```

Comment:
This current-implementation-vs-desire is a little odd in a design doc - feels like more of a TODO?
A parenthetical "this _could_ also be ... for better efficiency" might be a better framing?

Reply:
removed